auth0/resource_auth0_guardian.go

```diff
@@ -254,6 +281,17 @@ func readGuardian(d *schema.ResourceData, m interface{}) error {
 	if err != nil {
 		return err
 	}
+
+	factors, err := api.Guardian.MultiFactor.List()
+	if err != nil {
+		return err
+	}
+	for _, v := range factors {
```

Comment:
Isn't it more readable to avoid the switch here? I think it could be just 
```go
if v.Name != nil && *v.Name == "email"{
    d.Set("email", v.Enabled)
}
```

Reply:
I went for `switch` here thinking that it'll be a bit more concise in terms of syntax once `one-time-password` & `recovery-code` get added. Would you say you feel strongly about this point re: the readability aspect? If so then happy to change :).

Reply:
I see what you mean, I think we can leave the switch there 👍  But isn't there still a possible null pointer error here that is not adressed by your code? If `v.Name == nil` that is

Reply:
I'd agree that @apamildner 's if statement is slightly clearer. If and when one-time-password and recovery-code get added, a switch will likely make more sense.